pkg/features/features.go

```diff
@@ -49,6 +49,10 @@ type cfg struct {
 				Enabled bool `json:"enabled" yaml:"enabled" config:"enabled"`
 			} `json:"tamper_protection,omitempty" yaml:"tamper_protection,omitempty" config:"tamper_protection,omitempty"`
 		} `json:"features" yaml:"features" config:"features"`
+		// Agent config has protection as a top level field.
+		Protection *struct {
+			Enabled bool `json:"enabled" yaml:"enabled" config:"enabled"`
+		} `json:"protection,omitempty" yaml:"protection,omitempty" config:"protection,omitempty"`
```

Comment:
The initial attempt was always returning true when `TamperProtect()` was being called (default behaviour when the key is missing).
However, I found that the config being used in `features.Apply` did not match the expected struct defined here.
The struct expects `agent.features.tamper_protect.enabled`. However the config being passed had:
```yml
agent:
  protection:
    enabled: false
```
I added this additional optional struct as a fallback in case `features.tamper_protection` was not defined so loading the features in the `Uninstall` command would be able to parse the value